test/scenario/virtual_currency_test.cpp

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+#include <infra/protobuf/api.grpc.pb.h>
+#include <memory>
+#include <repository/domain/account_repository.hpp>
+#include <repository/domain/asset_repository.hpp>
+#include <service/executor.hpp>
+#include <unordered_map>
+
+Api::Account makeAccount(const std::string& publicKey, const std::string& name,
```

Comment:
this one and 2 fns below are code duping from `test/scenario/transfer_with_tax_test.cpp`